src/Containers/MiddleCanvas/DualSelectBar.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import styled from 'styled-components';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+
+
+export interface DualSelectBarProps extends MainInterface, ResponsiveInterface, React.HTMLAttributes<HTMLDivElement> {
+    leftSelectOption: string,
+    rightSelectOption: string,
+    selectedOption: string,
+    setSelectedOption: React.Dispatch<React.SetStateAction<string>>,
+};
+
+interface OptionProps {
+    isSelected?: boolean
+};
+
+const IS_SELECTED_COLOR = '#e70028';
+const UNSELECTED_COLOR = '#b7b7b7';
+
+export const DualSelectBar: React.FC<DualSelectBarProps> = ({
+    leftSelectOption,
+    rightSelectOption,
+    selectedOption,
+    setSelectedOption,
+    ...props
+}): React.ReactElement => {
+    return (
+        <Wrapper {...props}>
+            <Option 
+                isSelected={selectedOption === leftSelectOption}
+                onClick={() => setSelectedOption(leftSelectOption)}
+            >
```

Comment:
Abstract the duplciate logic with either an object spread
Function that returns an object spread

const getOptionFields(Left) getOptionFields(Right) => {
 return {
         isSelected={selectedOption === leftSelectOption}
                onClick={() => setSelectedOption(leftSelectOption)}
}
}